app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java

```diff
@@ -735,7 +735,13 @@ public void downloadTei(String teiUid, String enrollmentUid) {
                         .subscribe(
                                 view.downloadProgress(),
                                 Timber::d,
-                                () -> openDashboard(teiUid, enrollmentUid))
+                                () -> {
+                                    if(d2.trackedEntityModule().trackedEntityInstances().uid(teiUid).blockingExists()) {
+                                        openDashboard(teiUid, enrollmentUid);
+                                    }else{
+                                        view.couldNotDownload(trackedEntity.displayName());
+                                    }
+                                })
```

Comment:
I think searchTe has unit test. Do you think we could reflect this?

Reply:
Not possible now. Will do it later